tools/abel/validators/README.md

```diff
@@ -0,0 +1,192 @@
+
+`Entity Code` *string* **required**
+
+[Human-readable code](../../../ontology/docs/building_config.md#identifiers) for an entity e.g. `AHU-1001`
+
+`Entity Guid` *string*
+
+[UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random))
+[identifier](../../../ontology/docs/building_config.md#identifiers) for an entity. **Note:** If an entity belongs to the `FACITITLIES` namespace and ABEL is
+being used or Google, then this field is **required** or else spreadsheet
+validation will fail.
+
+`Etag` *string*
+
+Version control tag for entities in an entity type database. Used only when updating an existing building config.
+
+`Is Reporting` *Boolean* **required**
+
+`TRUE` if an entity is [reporting its own data](../../../ontology/docs/building_config.md#reporting-physical-devices) and `FALSE` if [the entity is virtual](../../../ontology/docs/building_config.md#virtual-devices).
+
+`Cloud Device ID` *string*
+
+Cloud registry [device numeric identifier](../../../ontology/docs/building_config.md#identifiers). The device numeric ID is automatically created by a Cloud IoT application, and it is globally unique and not editable.
+
+`DBO Namespace` *string* **required**
+
+[Namespace](../../../ontology/docs/ontology.md#namespaces) for an entity as defined in the [Digital Buidings Ontology](../../../ontology/README.md). e.g. `HVAC`
+
+`DBO General Type`
+
+[General Type](../../../ontology/docs/model.md#general-types) for an entity as defined in the [Digital Buildings Ontology](../../../ontology/README.md). e.g. `AHU`
+
+`DBO Entity Type Name` *string* **required**
+
+[Entity Type Name](../../../ontology.md#type-names) for an Entity as defined in the [Digital Buildings Ontology](../../../ontology/README.md). e.g. `VAV_SD_DSP`
+
+### Entity Fields
+
+The `Entity Fields` tab defines fields translated by [reporting entities](../../../ontology/docs/building_config.md#reporting-physical-devices) and linked to [virtual entities](../../../onotology/docs/building_config.md#virtual-devices) defined in the `Entities` tab.
+
+`Entity Code` *string* **required**
+
+A [human-readable code](../../../onotology/docs/building_config.md#identifiers) for an entity which an entity field is a data point of. This code must already be defined in the [Entities](#entities) table. e.g. `AHU-1001`
+
+`Entity Guid` *string*
+
+[UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) value for an entity corresponding to the entity in the Entity Code column ad is a unique [identifier for an entity](../../../onotology/docs/building_config.md#identifiers). This column need not be filled in unless the corresponding entity code is **not unique**. This column will be used when ABEL writes to a blank spreadsheet.
```

Comment:
`must not be filled`?

Reply:
Technically it can be if the user really wants to provide their own GUIDs but it's complete unnecessary except for in the duplicate code case.